pkg/autoupdate/tuf_autoupdate.go

```diff
@@ -0,0 +1,211 @@
+package autoupdate
+
+// This new autoupdater points to our new TUF infrastructure, and will eventually supersede
+// the legacy `Updater` in autoupdate.go that points to Notary.
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/version"
+	legacytuf "github.com/kolide/updater/tuf"
+	client "github.com/theupdateframework/go-tuf/client"
+	filejsonstore "github.com/theupdateframework/go-tuf/client/filejsonstore"
+)
+
+//go:embed assets/tuf-dev/root.json
+var rootJson []byte
+
+const (
+	DefaultTufServer = "https://tuf-devel.kolide.com"
+)
+
+type TufAutoupdater struct {
+	metadataClient  *client.Client
+	mirrorClient    *http.Client
+	mirrorUrl       string
+	binary          string
+	operatingSystem string
+	channel         UpdateChannel
+	checkInterval   time.Duration
+	errorCounter    []int64
+	lock            sync.RWMutex
+	interrupt       chan struct{}
+	logger          log.Logger
+}
+
+type TufAutoupdaterOption func(*TufAutoupdater)
+
+func WithTufLogger(logger log.Logger) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		ta.logger = log.With(logger, "component", "tuf_autoupdater")
+	}
+}
+
+func WithChannel(channel UpdateChannel) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		ta.channel = channel
+	}
+}
+
+func WithUpdateCheckInterval(checkInterval time.Duration) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		ta.checkInterval = checkInterval
+	}
+}
+
+func NewTufAutoupdater(metadataUrl, mirrorUrl, binaryPath, rootDirectory string, opts ...TufAutoupdaterOption) (*TufAutoupdater, error) {
+	binaryName := filepath.Base(binaryPath)
```

Comment:
Why pass `binaryPath` and not binaryName directly?

Is this going to need to strip off `.exe` on windows?

Reply:
Maybe? I think we should change where we store updates. We'll need to for nix, and the current idiom is kinda wonky

Reply:
I think per usual my worry is about changing too many preexisting things that work as-is 🙂 I'll update this to be the binaryName instead of path, and punt this down the road a bit.